cose/cose.go

```diff
@@ -0,0 +1,373 @@
+package cose
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"errors"
+	"io"
+	"reflect"
+
+	dtcbor "github.com/datatrails/go-datatrails-common/cbor"
+	"github.com/datatrails/go-datatrails-common/logger"
+	"github.com/fxamacker/cbor/v2"
```

Comment:
why do we need to import the external dependency cbor  as well as the go-datatrails-common cbor?

Reply:
they do different things. There is no need to hid the actual cbor under a datatrails cbor. we are not abstracting. and I don't think the attempts we have made in other places to do that have worked out well.